### ISSUE TITLE
rows and columns appear switched

### ISSUE THREAD

**Issue by m-schoenitz** [rharkes/Fast_Tiff_Write#2]
It's fast and mostly reliable, it helps a lot for writing large image stacks. Thanks.

However, the written images appear to be rotated and flipped (x and y axes are swapped)

Here is a bit of code to show this:
```
% fast Tiff test

testImg = [1,2;3,4;5,6];
testImg = im2uint16(rescale(testImg));

imwrite(testImg,'tiff-imwrite.tif');

fTIFF = Fast_Tiff_Write('tiff-fTIFF.tif',1,0);
fTIFF.WriteIMG(testImg); %% this does not work for some reason
% fTIFF.WriteIMG(permute(testImg,[2,1])); %% this does work
fTIFF.close;

subplot(1,3,1);
imagesc(testImg);
axis equal tight
title('original image')

subplot(1,3,2)
imagesc(imread('tiff-imwrite.tif'));
axis equal tight
title('read after imwrite')

subplot(1,3,3)
imagesc(imread('tiff-fTIFF.tif'));
axis equal tight
title('read after fTIFF.WriteIMG')
```
This produces this result:
![image](https://user-images.githubusercontent.com/107219834/210455495-3a4b73cd-9169-4e70-8a79-d04cc0cbd990.png)
In case OS and Matlab version are relevant: '9.6.0.1214997 (R2019a) Update 6' on 'PCWIN64' (Windows 10).

A simple workaround is to permute the image before writing (as in the commented-out line 10), but perhaps there is a way to not have to do this.

(Or perhaps I'm missing something obvious, that's also quite possible).

**Comment by rharkes** [comment]
Hi @m-schoenitz ,

Thanks for creating this issue. I'll have a look this week. It has been a while since I worked on this code, but I seem to remember a different convention about first and second dimension indeed.

**Comment by rharkes** [comment]
Today I looked into it. Two things I've found:
* Matlab uses the FORTRAN (column major) way of storing data ([ref](https://nl.mathworks.com/help/coder/ug/what-are-column-major-and-row-major-representation-1.html)). So that is how data [is written to the file](https://github.com/rharkes/Fast_Tiff_Write/blob/master/Fast_BigTiff_Write.m#L148). I'd rather not change that since it is probably partly responsible for the writing speed, but might have to.
* The default orientation in tiff is row first, as set in [tag 274](https://www.awaresystems.be/imaging/tiff/tifftags/orientation.html). It would be easy to set that tag to column first (5), however, I've not found a single reader that actually uses that tag...

So in conclusion, I should probably just permute the image data before writing so that it is written row-first.

**Comment by rharkes** [comment]
https://github.com/rharkes/Fast_Tiff_Write/tree/col-%3Erow-major

Could you check this branch?

**Comment by m-schoenitz** [comment]
Hey thanks for looking at this, sorry I've been slow.

Yes, this works now, at least to the extent I've found it to be not working previously. I haven't tested with color images.

However you do have some encoding problem I think. I only got it to work after opening it in a text editor and force-converting it to UTF8. Before that it threw an error 

```
>> fast_tiff_test
The file
   "D:\functions\Fast_Tiff_Write.m"
cannot be executed.
Error in fast_tiff_test (line 8)
fTIFF = Fast_Tiff_Write('tiff-fTIFF.tif',1,0); 

```


The file in Raw view - 
- in the master branch:
![image](https://user-images.githubusercontent.com/107219834/226970287-67ba670c-0bac-4378-a443-ee8df9897aa2.png)

- in this new branch:
![image](https://user-images.githubusercontent.com/107219834/226970366-f81f4f09-1d25-4974-a7e5-8d642ac888ed.png)

**Comment by rharkes** [comment]
Hey! No worries, also a slow response from my side unfortunately. But good to hear that it works for the images you use. I'll run a quick test on RGB images, make the changes in big_tiff_writer, and merge the branch. 

Thanks for finding that encoding problem. I see that indeed it has changed those three (c) characters in the commit. Will try to revert that.
